src/monstergenerator.cpp

```diff
@@ -658,6 +658,12 @@ void mtype::load( const JsonObject &jo, const std::string &src )
 
     optional( jo, was_loaded, "description", description );
 
+    if( !origin.empty() && origin.back().first != id ) {
+        origin.clear();
+    }
+    origin.emplace_back( id, mod_id( src ) );
+    assign_src( id, src );
```

Comment:
Shouldn't all this be `assign_src( *this, src )`?

Reply:
Turns out they were both unecessary, just needed to adjust the name since it seems monsters are loaded with `generic_factory`. Template magic is very scary.